service/kernel/data_management/finiteness_checker.cpp

```diff
@@ -0,0 +1,368 @@
+            for (size_t j = 0; j < nPerInstr; ++j)
+            {
+                if (res64_1[j] == 0x7ff0000000000000uLL)
+                    if (res64_2[j] != 0x0000000000000000uLL && allowNaN)
+                        finite[start + nPerInstr * i + j] = true;
+                    else
+                        finite[start + nPerInstr * i + j] = false;
+                else
+                    finite[start + nPerInstr * i + j] = true;
+            }
+        }
+        size_t offset = start + (lcSize / nPerInstr) * nPerInstr;
+        valuesAreFinite(dataPtrs[ptrIdx] + offset, finite + offset, end - offset, allowNaN);
+    });
+}
+
+template <typename DataType>
+bool checkFinitenessAVX512Impl(size_t nElements, size_t nDataPtrs, size_t nElementsPerPtr, DataType ** dataPtrs, bool allowNaN)
+{
+    size_t nBlocksPerPtr = nElementsPerPtr / BLOCK_SIZE;
+    if (nBlocksPerPtr == 0) nBlocksPerPtr = 1;
+    bool inParallel     = !(nElements < THREADING_BORDER);
+    size_t nPerBlock    = nElementsPerPtr / nBlocksPerPtr;
+    size_t nSurplus     = nElementsPerPtr % nBlocksPerPtr;
+    size_t nTotalBlocks = nBlocksPerPtr * nDataPtrs;
+    size_t optArrLen    = nElements + (64 - nElements % 64);
+    daal::services::internal::TArray<bool, avx512> finiteArr(optArrLen);
+    bool * finite              = finiteArr.get();
+    uint64_t * uint64FinitePtr = (uint64_t *)finite;
+    for (size_t i = 0; i < optArrLen / 64; ++i) uint64FinitePtr[i] = 0xffffffffffffffffuLL;
+
+    checkFinitenessInBlocks(dataPtrs, finite, inParallel, nTotalBlocks, nBlocksPerPtr, nPerBlock, nSurplus, allowNaN);
+
+    for (size_t i = 0; i < optArrLen / 64; ++i)
+        if (uint64FinitePtr[i] != 0xffffffffffffffffuLL) return false;
+
+    return true;
+}
+
+template <>
+bool checkFiniteness<float, avx512>(size_t nElements, size_t nDataPtrs, size_t nElementsPerPtr, float ** dataPtrs, bool allowNaN)
+{
+    return checkFinitenessAVX512Impl<float>(nElements, nDataPtrs, nElementsPerPtr, dataPtrs, allowNaN);
+}
+
+template <>
+bool checkFiniteness<double, avx512>(size_t nElements, size_t nDataPtrs, size_t nElementsPerPtr, double ** dataPtrs, bool allowNaN)
+{
+    return checkFinitenessAVX512Impl<double>(nElements, nDataPtrs, nElementsPerPtr, dataPtrs, allowNaN);
+}
+
+#endif
+
+template <typename DataType, daal::CpuType cpu>
+void allValuesAreFiniteImpl(NumericTable & table, bool allowNaN, bool * finiteness)
+{
+    const size_t nRows     = table.getNumberOfRows();
+    const size_t nColumns  = table.getNumberOfColumns();
+    const size_t nElements = nRows * nColumns;
+    const NTLayout layout  = table.getDataLayout();
+
+    size_t nDataPtrs, nElementsPerPtr;
+    if (layout == NTLayout::soa)
+    {
+        // SOA layout: pointer for each column
+        nDataPtrs       = nColumns;
+        nElementsPerPtr = nRows;
+    }
+    else
+    {
+        // AOS layout: one pointer for all data
+        nDataPtrs       = 1;
+        nElementsPerPtr = nRows * nColumns;
```

Comment:
SDL check should be added for possible integer overflow.

Reply:
use DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION macro for it.
see example: https://github.com/intel/daal/blob/master/algorithms/kernel/objective_function/cross_entropy_loss/cross_entropy_loss_dense_default_batch_impl.i#L191

Reply:
Added